src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSync.scala

```diff
@@ -261,6 +284,45 @@ class RegularSync(
     }
   }
 
+  private def requestMissingNode(nodeId: ByteString): Unit = {
+    bestPeer match {
+      case Some(peer) =>
+        waitingForActor = Some(context.actorOf(
+          PeerRequestHandler.props[GetNodeData, NodeData](
+            peer, peerResponseTimeout, etcPeerManager, peerEventBus,
+            requestMsg = GetNodeData(List(nodeId)),
+            responseMsgCode = NodeData.code)))
+
+      case None =>
+        log.debug("Requesting missing state nodes: no peers to download from")
+        scheduleResume()
+    }
+  }
+
+  private def handleRedownloadedStateNodes(peer: Peer, nodes: Seq[ByteString]): Unit = {
+    if (nodes.isEmpty) {
+      log.debug(s"Did not receive missing state node from peer ($peer)")
+      resumeWithDifferentPeer(peer, "did not receive missing state node")
+    } else {
+      val MissingStateNodeRetry(requestedHash, blockPeer, blocksToRetry) = missingStateNodeRetry.get
+      val receivedHash = kec256(nodes.head)
```

Comment:
it probably doesn't matter anyway, but how about `receivedNodeOpt = nodes.find(kec256(_) == requestedHash)` and then match on `receivedNodeOpt`?

Reply:
I don't think I follow, what is the gain here? (for both suggestions)

Reply:
for my suggestion, as I said it probably doesn't matter :) the gain is, that if we somehow receive 2 or more nodes we can still handle it. Anyway, you can disregard that comment

Reply:
And I meant that we asked for a single MPT node, why would respond with more than 1?

Reply:
> why would respond with more than 1

exactly, I chose not to deal with other situations.